azure-pipelines.release-fluentui.yml

```diff
@@ -1,74 +1,220 @@
+        displayName: Checkout branch
+        inputs:
+          script: >-
+            git checkout $(Build.SourceBranchName)
+
+            git pull
+
+      - task: NodeTool@0
+        displayName: Install Node.js
+        inputs:
+          versionSpec: 12.x
+
+      - task: YarnInstaller@0
+        displayName: Use Yarn 1.19.x
+        inputs:
+          versionSpec: 1.19.x
+          checkLatest: true
+
+      - task: Bash@3
+        displayName: Yarn
+        inputs:
+          filePath: yarn-ci.sh
+
+      - task: CmdLine@2
+        displayName: yarn buildci
+        inputs:
+          script: yarn buildci
+
+      - task: CmdLine@2
+        displayName: '[Nightly] Pack fluentui packages'
+        condition: and(succeeded(), eq(variables.publishOfficial, false))
+        inputs:
+          script: >-
+            today=`date +%Y-%m-%d`
+
+            echo date today $today
+
+            echo "##vso[task.setvariable variable=date]$today"
+
+            yarn release:fluentui:pack-nightly
+
+      - task: AzureUpload@2
+        displayName: '[Nightly] Upload tarballs'
+        condition: and(succeeded(), eq(variables.publishOfficial, false))
+        inputs:
+          SourcePath: fluentui-nightly
+          ConnectedServiceNameARM: 4f791194-9458-4a18-9f9d-6a77ddb74b8c
+          StorageAccountRM: fluentsite
+          ContainerName: nightly-builds
+          # upload tarballs to folder named by today's date
+          BlobPrefix: $(date)
+          Gzip: true
+
+      - task: AzureUpload@2
+        displayName: '[Nightly] Upload tarballs as newest'
+        condition: and(succeeded(), eq(variables.publishOfficial, false))
+        inputs:
+          SourcePath: fluentui-nightly
+          ConnectedServiceNameARM: 4f791194-9458-4a18-9f9d-6a77ddb74b8c
+          StorageAccountRM: fluentsite
+          ContainerName: nightly-builds
+          # upload tarballs to folder named 'newest'. This folder is used by the nightly version of docsite to create codesandbox
+          BlobPrefix: newest
+          Gzip: true
+
+      - task: CmdLine@2
+        displayName: '[NPM] Publish to NPM'
+        condition: and(succeeded(), eq(variables.publishOfficial, true))
+        timeoutInMinutes: 2
+        inputs:
+          script: >-
+            yarn logout
+
+            export NPM_TOKEN=$(npmToken)
+
+            echo "@fluentui:registry=http://registry.npmjs.org/" > .npmrc
+
+            echo "registry=http://registry.npmjs.org/" >> .npmrc
+
+            echo //registry.npmjs.org/:_authToken=$(npmToken) >> .npmrc
+
+            touch packages/fluentui/.npmrc
+
+            echo "@fluentui:registry=http://registry.npmjs.org/" > packages/fluentui/.npmrc
+
+            echo "registry=http://registry.npmjs.org/" >> packages/fluentui/.npmrc
+
+            echo //registry.npmjs.org/:_authToken=$(npmToken) >> packages/fluentui/.npmrc
+
+            yarn release:fluentui:$(version) --yes
+
+            yarn release:fluentui:post-validation
```

Comment:
I don't see anything similar in `azure-pipelines.release.yml`, why this is required?

Reply:
because we don't use beachball which does some magic behind the scenes that more or less sets the registry url with token

Reply:
@ecraig12345 do you have better ideas?

Reply:
We could do similar to this and manually set the `--registry` arg in lerna instead of writing the .npmrc to disk:

https://github.com/microsoft/beachball/blob/master/src/packageManager/packagePublish.ts

Reply:
@layershifter Not sure sorry, my experience configuring publishing is pretty limited

Reply:
> We could do similar to this and manually set the `--registry` arg in lerna instead of writing the .npmrc to disk:
> 
> https://github.com/microsoft/beachball/blob/master/src/packageManager/packagePublish.ts

Not sure if this will work, I remember I need both .npmrc in root and in fluent folder for lerna to work.